test/controllers/categories_controller_test.rb

```diff
@@ -10,35 +10,81 @@
     
     it "gives back a 404 if there are no works available" do
       get categories_path
-      
+      must_respond_with :success
     end
   end
   
-  describe 'show action' do
-    it 'responds with a success when id given exists' do
-      valid_work = Category.first
+  
+  describe 'create action' do
+    
+    it 'should not let the guest user to create a new category, no data' do
       
-      get category_path(valid_category.id)
+      post categories_path, params: {
+        category: {name: nil}
+      }
       
-      must_respond_with :success
+      must_respond_with :redirect #flash
+    end
+    
+    it 'should not let the guest user to create a new category' do
+      category_hash = {
+        category: { name: "Flowers"}
+      }
+      expect {
+        post categories_path, params: category_hash
+      }.must_differ 'Category.count', 0
+      must_redirect_to root_path
+    end
+    
+    it 'should let the merchant to create a new  category' do
+      category_hash = {
+        category: { name: "Flowers"}
+      }
+      expect {
+        post categories_path, params: category_hash
+      }.must_differ 'Category.count', 1
+      must_redirect_to root_path
+    end
+  end
+  
+  describe 'update action' do
+    before do
+      @new_category = Category.create(name: "new category")
```

Comment:
This may be a discussion for tomorrow but what is this @new_category for? I'm not seeing it used anywhere else in your code so I'm curious as to what the purpose of it is?